sdk/core/azure-core/src/main/java/com/azure/core/util/Configuration.java

```diff
@@ -247,7 +272,12 @@ public static Configuration getGlobalConfiguration() {
      * @return Value of the configuration if found, otherwise null.
      */
     public String get(String name) {
-        return getOrLoad(name);
+        return getOrLoad(new ConfigurationProperty(name));
```

Comment:
I get a little nervous about this in terms of GC churn if we are checking configuration frequently (I'm not sure if we have cases like this or not). In these retrofitted cases, we are still using the exact same name, but it is just boxed and unboxed in a ConfigurationProperty?

Reply:
sorry, this is prototype and subject for optimization - I was going to create it as a draft - nothing is final here :)

Reply:
Understood - no pressure! :-)